scripts/ghprb_deduper.py

```diff
@@ -0,0 +1,237 @@
+                ** number: the build id number
+                ** actions: a list of 'actions', from which the only
+                    item used in this script is 'parameters' which can
+                    be used to find the PR number.
+        """
+        api_url = self._append_url(self.job_url, '/api/json')
+
+        response = requests.get(
+            api_url,
+            params={
+                "tree": ("builds[building,timestamp,"
+                         "number,actions[parameters[*]]]"),
+            }
+        )
+
+        return response.json()
+
+    def get_running_builds(self, data):
+        """
+        Return build data for currently running builds.
+
+        :Args:
+            data: the return value of self.get_json()
+
+        :Returns:
+            build_data: a dict where keys are a PR id and the value is
+                a list of 'builds' for the PR. Each item in the list
+                has the data for the build as returned in self.get_json.
+        """
+        build_data = defaultdict(list)
+
+        # This is used to find the PR id instead of having a bunch of
+        # messy loops.
+        pr_id_regex = (r"{[\s-]*\"name\"[\s-]*:[\s-]*\"ghprbPullId\""
+                       r"[\s-]*,[\s-]*\"value\"[\s-]*:[\s-]*\"[\d]*\"}")
+
+        for b in data['builds']:
+            if b['building']:
+                pr_id_data_str = re.search(
+                    pr_id_regex, json.dumps(b['actions'])).group(0)
+                pr_id_data = json.loads(pr_id_data_str)
+                pr_id = pr_id_data['value']
+                build_data[pr_id].append(b)
+
+        return build_data
+
+    def stop_duplicates(self, build_data):
+        """
+        Finds PRs that have multiple builds actively running.
+
+        :Args:
+            build_data: the data returned by self.get_running_builds()
+        """
+        lines = []
+        for pr, builds in build_data.iteritems():
+            if len(builds) > 1:
+                # Sort builds by timestamp (newest to oldest)
+                sorted_builds = sorted(
+                    builds, key=itemgetter('timestamp'), reverse=True)
+
+                # Most recently started build is the first in the list
+                current_build_id = sorted_builds[0]['number']
+
+                # Any other builds are assumed to be outdated
+                old_build_ids = [b['number'] for b in sorted_builds[1:]]
+
+                lines.append("PR #{}:".format(pr))
+                lines.append("\tNum running builds: {}".format(
+                    len(sorted_builds)))
+                lines.append("\tCurrent build: {}".format(
+                    current_build_id))
+                lines.append("\tOutdated builds: {}".format(
+                    old_build_ids))
+
+                desc = self._aborted_description(current_build_id, pr)
+
+                for b in old_build_ids:
+                    try:
+                        self.update_build_desc(b, desc)
```

Comment:
If the build finished up while we were calculating which ones to abort, then this will update the build desciption saying that we stopped it when we didn't. That might be confusing.
I'm imagining that you can update the build description after stopping it tho, so maybe switch these two line and if the aborting of the build is caught then the desc won't be updated.
This is assuming that if you call stop and the build is no longer running, it will get caught as an exception. Not sure if that's true.